tv_apps/smart-home/js/card_picker.js

```diff
@@ -38,16 +41,69 @@
       this.container.addEventListener('keyup', this.onKeyUp.bind(this), true);
 
       this._selectedElements = this.gridView.getElementsByClassName('selected');
+
+      this.input.addEventListener('focus', () => {
+        setTimeout(() => {
+          this.input.setSelectionRange(
+            this.input.value.length, this.input.value.length);
+        }, 0);
+      });
+    },
+
+    openKeyboard: function () {
+      this.panel.classList.add('hidden');
+      this.input.readOnly = false;
+      this.input.blur();
+      this.input.focus();
+      // There is a case that
+      // when user long presses Space key on virtual keyboard,
```

Comment:
This behavior only happens on native keyboard app, not partner's. So I suggest add some general comments here like: "Force to close the keyboard in case the keyboard is dismissed by other reasons."